### ISSUE TITLE
Feat: Support secrets in compose file v3

### ISSUE THREAD

**Issue by cghislai** [docker/docker#29968]
Now that secret managment is available, shouldn't we be able to reference them from a compose file? I guess secrets should always be external and the stack should fail to deploy when some secrets are missing.
I don't see anything about that point on the bug trackers.

**Comment by justincormack** [comment]
Sorry, how would you want to specify them in compose?

**Comment by cghislai** [comment]
@justincormack I was thinking of something like
```
services:
  service:
    secrets: 
     - service_credential
     - service_credential2

secrets:
  service_credential: {}
  service_credential2:
    external: true
```
Now im not sure if its better to force secrets to be created externally or to remains consistent with volumes/networks behaviour.

**Comment by brunoqc** [comment]
Shouldn't this issue be closed since its implemented by #30144?

**Comment by thaJeztah** [comment]
@brunoqc yes, thanks for that! Looks like the PR didn't include a "fixes #29968", so it didn't auto close.

This is indeed implemented and part of the docker 1.13.1 release
